fraos/src/seqno_iter.rs

```diff
@@ -14,24 +15,22 @@ impl SeqNoIter {
         Self { data, index, seqno }
     }
 
-    fn next_impl(&mut self) -> Option<Vec<u8>> {
-        let offset = self.index.get_pointer_to_value(self.seqno)? as usize;
-        let next_offset = self
-            .index
-            .get_pointer_to_value(self.seqno + 1)
-            .map(|value| value as usize)
-            .unwrap_or_else(|| self.data.memory_size());
-        let length = next_offset - offset;
+    pub fn next_impl(&mut self) -> Result<Option<Vec<u8>>, FraosError> {
+        let (offset, length) = match self.index.get_offset_and_length(self.seqno)? {
+            None => return Ok(None),
+            Some((offset, length)) => (offset, length),
+        };
+
         let item = self.data.get_record_at_offset(offset, length)?;
         self.seqno += 1;
-        Some(item)
+        Ok(item)
     }
 }
 
 impl Iterator for SeqNoIter {
     type Item = Vec<u8>;
 
     fn next(&mut self) -> Option<Self::Item> {
-        self.next_impl()
+        self.next_impl().unwrap()
```

Comment:
why not change the item to `Result`? backwards compatibility?

Reply:
in that case we won't be able to do sth like `for i in iter`. do you know if there is sth like `try_iter` or sth similar which can return result and support `in`?

Reply:
Well, you can usually do:

```
for i in iter {
     let i = i?;
}
```

Which I generally don't mind doing.